src/uberComponents/DetailPage/index.tsx

```diff
@@ -0,0 +1,97 @@
+import { Bundle, Resource } from 'fhir/r4b';
+import { useEffect, useState } from 'react';
+import { useParams, Route, Routes, useLocation, Link, useNavigate, Params } from 'react-router-dom';
+
+import { SearchParams, RenderRemoteData, useService, WithId } from '@beda.software/fhir-react';
+
+import { PageContainer } from 'src/components';
+import { RouteItem } from 'src/components/BaseLayout/Sidebar/SidebarTop';
+import { Tabs } from 'src/components/Tabs';
+import { getFHIRResources } from 'src/services';
+import { compileAsFirst } from 'src/utils';
+
+export interface Tab<R extends Resource> {
+    label: string;
+    path?: string;
+    component: (resource: R) => JSX.Element;
+}
+
+interface DetailPageProps<R extends Resource> {
+    resourceType: R['resourceType'];
+    getSearchParams: (params: Readonly<Params<string>>) => SearchParams;
+    getTitle: (resource: R) => string | undefined;
+    tabs: Array<Tab<R>>;
+    basePath: string;
+    extractPrimaryResource?: (bundle: Bundle<R>) => R;
+}
+
+interface PageTabsProps<R extends Resource> {
+    tabs: Array<Tab<R>>;
+    basePath: string;
+}
+
+export function PageTabs<R extends Resource>({ tabs, basePath }: PageTabsProps<R>) {
+    const location = useLocation();
+    const params = useParams<{ id: string }>();
+    const navigate = useNavigate();
+
+    const menuItems: RouteItem[] = tabs.map(({ label, path }) => ({
+        label,
+        path: `/${basePath}/${params.id}/${path}`,
+    }));
+
+    const [currentPath, setCurrentPath] = useState(location?.pathname);
```

Comment:
I don't think we should store currentPath in the state if we have location and we subscribed to it using useLocation. Just use location.pathname instead of currentPath